File: Serial2.java
public class Serial2 {
	
	public static void main(String args[]) throws InterruptedException {
		Serial2 t = new Serial2();
		System.out.println("Start!");
		long startTime = System.currentTimeMillis();
		t.testCase1();
		t.testCase2();
		t.testCase3();
		t.testCase4();
		long endTime = System.currentTimeMillis();
		long wait = ((endTime- startTime)/1000);
		System.out.println("End!");
		System.out.println("Time Taken : "+wait+" sec" );
	}
	
	public void testCase1() throws InterruptedException{
		System.out.println("Start 1");
		Thread.sleep(1000);
		System.out.println("End 1");
	}

	public void testCase2()throws InterruptedException {
		System.out.println("Start 2");
		Thread.sleep(2000);
		System.out.println("End 2");
	}
	
	public void testCase3()throws InterruptedException {
		System.out.println("Start 3");
		Thread.sleep(3000);
		System.out.println("End 3");
	}
	public void testCase4()throws InterruptedException {
		System.out.println("Start 4");
		Thread.sleep(4000);
		System.out.println("End 4");
	}
}

File: Serial2.py
import time
print("Start")
def Testcase1():
    start_time = time.time()
    print("Test case 1")
    time.sleep(1)
    end_time = time.time()
    print("total time for test case1 ",round(end_time-start_time))
    return
def Testcase2():
    start_time = time.time()
    print("Test case 2")
    time.sleep(2)
    end_time = time.time()
    print("total time for test case2 ",round(end_time-start_time))
    return
def Testcase3():
    start_time = time.time()
    print("Test case 3")
    time.sleep(3)
    end_time = time.time()
    print("total time for test case3 ",round(end_time-start_time))
    return
def Testcase4():
    start_time = time.time()
    print("Test case 4")
    time.sleep(4)
    end_time = time.time()
    print("total time for test case4 ",round(end_time-start_time))
    return
start_time = time.time()
t1 = Testcase1()
t2 = Testcase2()
t3 = Testcase3()
t4 = Testcase4()
end_time = time.time()
print("total time for all testcases ",round(end_time-start_time), "seconds")
print("Done")

